src/query/vacuum.c

```diff
@@ -4178,7 +4178,19 @@ vacuum_data_load_and_recover (THREAD_ENTRY * thread_p)
 	   * in the log_Gl header. After a long session in SA_MODE, the vacuum_Data.last_page->data->blockid will
 	   * be outdated. Instead, SA_MODE updates log_Gl.hdr.vacuum_last_blockid before removing old archives.
 	   */
-	  vacuum_Data.set_last_blockid (MAX (log_Gl.hdr.vacuum_last_blockid, vacuum_Data.last_page->data->blockid));
+	  VACUUM_LOG_BLOCKID last_blockid;
+
+	  if (vacuum_Data.is_restoredb_session)
+	    {
+	      last_blockid = vacuum_Data.last_page->data->blockid;
+	      log_Gl.hdr.vacuum_last_blockid = 0;
```

Comment:
How about put this initialization code into vacuum_initialize()

when the argument 'is_restore' is true, the set the log_Gl.hdr.vacuum_last_blockid to 0.

`int vacuum_initialize (...., bool is_restore)`
`{`
`   if (is_restore)` 
`    {`
`        log_Gl.hdr.vacuum_last_blockid = 0;`
`    }`
`}`

I think it is better to set vacuum_last_blockid when restoredb is executed.
If log_Gl.hdr.vacuum_last_blockid is set 0 when restoredb is executed, then original code is not required to be modified 
`vacuum_Data.set_last_blockid (MAX (log_Gl.hdr.vacuum_last_blockid, vacuum_Data.last_page->data->blockid));` 
what do you think?

Reply:
- If your feedback is to initialize log_Gl.hdr.vacuum_last_blockid at the time of restoredb, the code I modified is also initialized at the time of restoredb.
- It would be better to change log_Gl.hdr.vacuum_last_blockid after restoredb completes recovery in log_initialize().
- log_Gl.hdr.vacuum_last_blockid is only meaningful when vacuum_empty().

Reply:
OK, I understand. 
I didn't consider the `It would be better to change log_Gl.hdr.vacuum_last_blockid after restoredb completes recovery in log_initialize().`.